layers/layer_options.cpp

```diff
@@ -329,12 +329,63 @@ static bool SetBool(std::string &config_string, std::string &env_string, bool de
     return result;
 }
 
+static std::string GetConfigValue(const char *setting) { return getLayerOption((std::string("khronos_validation.") + setting).c_str()); }
+
+static std::string GetEnvVarValue(const char *setting) {
+    std::string env_var = setting;
+    std::transform(env_var.begin(), env_var.end(), env_var.begin(), ::toupper);
+    return GetEnvironment((std::string("VK_LAYER_") + env_var).c_str());
+}
+
+static std::optional<std::string> GetSettingValue(const char *setting) {
+    const std::string cfg_value = GetConfigValue(setting);
+    const std::string env_value = GetEnvVarValue(setting);
+
+    if (!env_value.empty()) {
+        return env_value;
+    } else if (!cfg_value.empty()) {
+        return cfg_value;
+    }
+    return {};
+}
+
+static void SetValidationSetting(CHECK_DISABLED &disable_data, const DisableFlags feature_disable, const char *setting) {
+    const std::optional<std::string> setting_value = GetSettingValue(setting);
+
+    if (setting_value) {
+        disable_data[feature_disable] = setting_value != "true";
+    }
+}
+
+static void SetValidationSetting(CHECK_ENABLED &enable_data, const EnableFlags feature_enable, const char *setting) {
+    const std::optional<std::string> setting_value = GetSettingValue(setting);
+
+    if (setting_value) {
+        enable_data[feature_enable] = setting_value == "true";
+    }
+}
+
+static void SetValidationGPUBasedSetting(CHECK_ENABLED &enable_data, const char *setting) {
+    const std::optional<std::string> setting_value = GetSettingValue(setting);
+
+    std::string list_of_flags;
+    std::string delimiter;
+    if (setting_value) {
+        list_of_flags = *setting_value;
+    }
+
+    if (!list_of_flags.empty()) {
+        enable_data[gpu_validation] = list_of_flags.find("GPU_BASED_GPU_ASSISTED") != std::string::npos;
+        enable_data[debug_printf] = list_of_flags.find("GPU_BASED_DEBUG_PRINTF") != std::string::npos;
+    }
```

Comment:
`delimiter` appears to be unused? And is there a need to copy `setting_value` to `list_of_flags`?
```suggestion
    const auto setting_value = GetSettingValue(setting);

    if (setting_value) {
        enable_data[gpu_validation] = setting_value->find("GPU_BASED_GPU_ASSISTED") != std::string::npos;
        enable_data[debug_printf] = setting_value->find("GPU_BASED_DEBUG_PRINTF") != std::string::npos;
    }
```